osu.Game/Database/MissingBeatmapNotification.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.Drawables.Cards;
+using osu.Game.Configuration;
+using osu.Game.IO.Archives;
+using osu.Game.Localisation;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Overlays.Notifications;
+using osu.Game.Scoring;
+using Realms;
+
+namespace osu.Game.Database
+{
+    public partial class MissingBeatmapNotification : SimpleNotification
+    {
+        [Resolved]
+        private BeatmapModelDownloader beatmapDownloader { get; set; } = null!;
+
+        [Resolved]
+        private ScoreManager scoreManager { get; set; } = null!;
+
+        [Resolved]
+        private RealmAccess realm { get; set; } = null!;
+
+        private readonly ArchiveReader scoreArchive;
+        private readonly APIBeatmapSet beatmapSetInfo;
+        private readonly string beatmapHash;
+
+        private Bindable<bool> autoDownloadConfig = null!;
+        private Bindable<bool> noVideoSetting = null!;
+        private BeatmapCardNano card = null!;
+
+        private IDisposable? realmSubscription;
+
+        public MissingBeatmapNotification(APIBeatmap beatmap, ArchiveReader scoreArchive, string beatmapHash)
+        {
+            beatmapSetInfo = beatmap.BeatmapSet!;
+
+            this.beatmapHash = beatmapHash;
+            this.scoreArchive = scoreArchive;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuConfigManager config)
+        {
+            realmSubscription = realm.RegisterForNotifications(
+                realm => realm.All<BeatmapSetInfo>().Where(s => !s.DeletePending), beatmapsChanged);
+
+            realm.Run(r =>
+            {
+                if (r.All<BeatmapSetInfo>().Any(s => !s.DeletePending && s.OnlineID == beatmapSetInfo.OnlineID))
+                {
+                    Text = NotificationsStrings.MismatchingBeatmapForReplay;
```

Comment:
Isn't this going to get overwritten by the `Text=` in `LoadComplete`?

Reply:
Yeah good spot. Should be addressed by https://github.com/ppy/osu/pull/24450/commits/ed9039f60f31415089be387e9075e6439099ee58.